jujugui/static/gui/src/app/state/state.js

```diff
@@ -82,6 +95,76 @@ const State = class State {
   }
 
   /**
+    Check the supplied block of path to see if this is a user store path or
+    not. These options are becuase the store path can come in many forms and
+    we need to know if it's a user store path for both parsing and
+    generating the path.
+    @return {Boolean}
+  */
+  _isUserStorePath(block) {
```

Comment:
I am not sure this function is useful. Why don't we keep the u/ prefix when parsing the URL? It seems that we are stripping it when saving the block in the state and then we need to add more logic just to check if state.store represents a user owned store entity. Why not considering u/ part of the charm or bundle URL (as it actually is)? At that point we would be able to just remove this function in theory?